### ISSUE TITLE
access to omemo contacts prints

### ISSUE THREAD

**Issue by FilLupin** [iNPUTmice/Conversations#4260]
Hi,
after a lock of database, I am unable to use my Conversations app (v2.9.3+fcr).
While finding a way to solve this, I would like to get the omemo keys from my contacts.

With sqlite DB-browser, I am able to access to Conversations tables.
Where can I access to omemo contacts prints?
Sorry since it is probably an often asked question, but I was not able to find the good keywords to find it on this repo.
 
Regards

**Comment by licaon-kter** [comment]
Their keys are on their server, thier of no use for you to move/copy/etc.

You might wanna care about your keys, but those are copied/moved during backup/restore.


With root you can copy the whole install and restore it (eg. with Oandbackupx) to a new device.

**Comment by FilLupin** [comment]
If I want to overcome the app bug with a computer software, I would like to get the omemo prints to avoid getting again all my keys.
Since I can enable an omemo print, it should be possible to get them through files, isn't it?

**Comment by licaon-kter** [comment]
The issues that you open were just questions better answered in the support channel, so better join it instead:

xmpp:conversations@conference.siacs.eu?join

**Comment by FilLupin** [comment]
Thank you for your answer.
However, I am not sure I will be able to join soon since my app is broken for now. Since my issue ould help me to overcome this app crash, is there another way (mail for example) to join you ?
Regards

**Comment by licaon-kter** [comment]
Install a fork like Conv6ations and make another account, then join. You're not stuck...

**Comment by FilLupin** [comment]
Thank you. 
Since I need a version compatible with android 4.2.2, do you know where I could find an APK 2.9.x?

**Comment by licaon-kter** [comment]
F-Droid...enable Archive repo...

**Comment by FilLupin** [comment]
Is there any way to get a version older than 3 last versions through a browsable website?
Thank you very much for your help.

**Comment by licaon-kter** [comment]
Yes...if you know the link. Why? Isn't F-Droid client working?

**Comment by FilLupin** [comment]
Yep, I already had some issues installing apk through f-droid, so I prefer avoiding adding another difficulty to this current ones. It will help me debugginh and isolating issues. ;)

**Comment by FilLupin** [comment]
For information of other users which need the info, I found the way to find an app through a browser : https://f-droid.org/archive/<APK name>, where <APK name> format can be found in the APK links of the app page on f-droid.org, for example Conv6ations last APK is named eu.sum7.conversations_42018.apk so I found an older APK on https://f-droid.org/archive/eu.sum7.conversations_42015.apk

**Comment by licaon-kter** [comment]
You'd need to grab a logcat of thu exact error, why did it crash?

**Comment by FilLupin** [comment]
Here is it
```
12-26 12:30:20.725 W/Trace   ( 5012): error opening trace file: No such file or directory (2)
12-26 12:30:20.725 W/InputMethodManagerService( 2262): Focus gain on non-focused client com.android.internal.view.IInputMethodClient$Stub$Proxy@41a8d3b0 (uid=10050 pid=4862)
12-26 12:30:20.760 W/PackageParser( 5012): /storage/sdcard1/app/Conv6ations/eu.sum7.conversations_42015.apk (at Binary XML file line #7): Requires newer sdk version #21 (current version is #17)
12-26 12:30:20.765 D/AndroidRuntime( 5012): Shutting down VM
12-26 12:30:20.765 W/dalvikvm( 5012): threadid=1: thread exiting with uncaught exception (group=0x416b6930)
12-26 12:30:20.770 E/AndroidRuntime( 5012): FATAL EXCEPTION: main
12-26 12:30:20.770 E/AndroidRuntime( 5012): java.lang.RuntimeException: Unable to start activity ComponentInfo{com.android.packageinstaller/com.android.packageinstaller.PackageInstallerActivity}: java.lang.NullPointerException
12-26 12:30:20.770 E/AndroidRuntime( 5012): 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2308)
12-26 12:30:20.770 E/AndroidRuntime( 5012): 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2358)
12-26 12:30:20.770 E/AndroidRuntime( 5012): 	at android.app.ActivityThread.access$600(ActivityThread.java:153)
12-26 12:30:20.770 E/AndroidRuntime( 5012): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1247)
12-26 12:30:20.770 E/AndroidRuntime( 5012): 	at android.os.Handler.dispatchMessage(Handler.java:99)
12-26 12:30:20.770 E/AndroidRuntime( 5012): 	at android.os.Looper.loop(Looper.java:137)
12-26 12:30:20.770 E/AndroidRuntime( 5012): 	at android.app.ActivityThread.main(ActivityThread.java:5227)
12-26 12:30:20.770 E/AndroidRuntime( 5012): 	at java.lang.reflect.Method.invokeNative(Native Method)
12-26 12:30:20.770 E/AndroidRuntime( 5012): 	at java.lang.reflect.Method.invoke(Method.java:511)
12-26 12:30:20.770 E/AndroidRuntime( 5012): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:795)
12-26 12:30:20.770 E/AndroidRuntime( 5012): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:562)
12-26 12:30:20.770 E/AndroidRuntime( 5012): 	at dalvik.system.NativeStart.main(Native Method)
12-26 12:30:20.770 E/AndroidRuntime( 5012): Caused by: java.lang.NullPointerException
12-26 12:30:20.770 E/AndroidRuntime( 5012): 	at android.content.pm.PackageParser.collectCertificates(PackageParser.java:596)
12-26 12:30:20.770 E/AndroidRuntime( 5012): 	at com.android.packageinstaller.PackageUtil.getPackageInfo(PackageUtil.java:75)
12-26 12:30:20.770 E/AndroidRuntime( 5012): 	at com.android.packageinstaller.PackageInstallerActivity.onCreate(PackageInstallerActivity.java:513)
12-26 12:30:20.770 E/AndroidRuntime( 5012): 	at android.app.Activity.performCreate(Activity.java:5104)
12-26 12:30:20.770 E/AndroidRuntime( 5012): 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1080)
12-26 12:30:20.770 E/AndroidRuntime( 5012): 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2262)
12-26 12:30:20.770 E/AndroidRuntime( 5012): 	... 11 more
12-26 12:30:20.770 W/ActivityManager( 2262):   Force finishing activity com.android.packageinstaller/.PackageInstallerActivity
12-26 12:30:20.775 W/WindowManager( 2262): Failure taking screenshot for (328x583) to layer 21065
12-26 12:30:20.775 W/ActivityManager( 2262):   Force finishing activity com.amaze.filemanager/.activities.MainActivity
12-26 12:30:21.275 W/ActivityManager( 2262): Activity pause timeout for ActivityRecord{41975ff8 u0 com.android.packageinstaller/.PackageInstallerActivity}
12-26 12:30:22.850 I/Process ( 5012): Sending signal. PID: 5012 SIG: 9
12-26 12:30:22.855 I/ActivityManager( 2262): Process com.android.packageinstaller (pid 5012) has died.
```
I imagine I could edit `AndroidManifest.xml` to manually change `MinSdkVersion` but it contains `\00\00` which seems strange and make it difficult to easily edit it (if you know a way to do it easily, I'm interested).

**Comment by licaon-kter** [comment]
https://f-droid.org/archive/eu.sum7.conversations_397.apk is tha latest compatible with Android 4

**Comment by FilLupin** [comment]
Thank you very much @licaon-kter, not only have I been able to access to my XMPP account but also did I access to my history!
Cheers

**Comment by licaon-kter** [comment]
Which one worked?

**Comment by FilLupin** [comment]
Conv6ations, but I can test blabber if it can help

**Comment by FilLupin** [comment]
Strange thing: in account, my omemo fingerprint seems to be the one of the Conversations account, and not the one seen by my contacts.
Do you know how to fix it?

**Comment by licaon-kter** [comment]
When you transfered your data, it contains your keys.

Contacts see how many keys?

**Comment by FilLupin** [comment]
I see :
- 1 key associated to my session, which is not the one seen by my contacts
- 3 keys associated to other devices. In these 3 keys, 2 can be (dis)trusted, and one is in fact the one seen by my contacts and I can't untrust it.

I imagined this bug is perhaps due to a copy of my OMEMO fingerprint just for display

I looked into DB contained in `history` file. 
I found in table `identities` entries with field `fingerprint`. For Conversations and Conv6ations clients, those keys begins with the same prefix and then follows the fingerprint displayed into my Conversations/Conv6ations clients.

The field `ownkey` is set to 1 only for the key of my former Conversations account, and the `trust`.field is set to `VERIFIED` (other fingerprints are associated to `TRUSTED`, `UNTRUSTED` or `UNDECIDED`).
I understand this ownkey should be set to 1 for the current client, and the Conversations fingerprint should be set to 0. Am I right?
